proto/config/netinst.proto

```diff
@@ -153,4 +153,21 @@ message NetworkInstanceConfig {
   // This option is only valid for local network instances. For other types
   // of network instances, it will be ignored.
   repeated IPRoute static_routes = 43;
+
+  // Maximum transmission unit (MTU) to set for the network instance and all application
+  // interfaces connected to it.
+  // This is the largest IP packet that the network instance is allowed to carry.
+  // This does not include the L2 header size (e.g. Ethernet header or a VLAN tag size).
+  // The value is a 16-byte unsigned integer, representing the MTU size in bytes.
+  // The minimum accepted value for the MTU is 1280 (RFC 8200, "IPv6 minimum link MTU").
+  // If not defined (zero value), EVE will set the MTU to the default value of 1500 bytes.
+  // On the host side, MTU is set to interfaces by EVE. On the guest (application) side,
+  // it is the role of the DHCP client to apply the MTU value, which it will receive with
+  // the DHCP lease via option 26.
```

Comment:
what is option 26?

Reply:
https://datatracker.ietf.org/doc/html/rfc2132#section-5.1

Reply:
@eriknordmark @uncleDecart @naiming-zededa 
Speaking of MTU on the guest-side (inside the app), I wonder how to propagate MTU to applications connected to switch NIs (we do not run DHCP/SLAAC for those). For container apps run inside our shim-VM, we could write per-interface MTU to the kernel cmdline and apply them from our init-initrd script (e.g. "mtu=1500,1500,9000" for eth0, eth1and eth2, respectively). The only downside, is that while MTUs of VIFs connected to local NIs will be allowed to change in-runtime by use of DHCP lease renew (or IPv6 RA update), MTUs of switch VIFs would stay fixed after the shim-VM boot (probably acceptable limitation).
However, I do not have any idea how to automatically propagate MTU of switch NI VIFs to VM applications.

Reply:
I would think pre-reboot setting should be good enough, we can say if the external network wants to change, then the external dhcp server is responsible to convey the mtu message.

Reply:
OK, that will do for **container** apps (where we initialize the shim-VM).
@naiming-zededa do you have some idea how we could automatically set MTUs of **VM** application interfaces connected to switch NIs? I believe this is completely out of our control, but maybe I'm missing something.

Reply:
I'm not familiar with that, it is possible it can only be set through the cloud-init during the first bootup.

Reply:
Is it correct that the issue appears for a VM (Linux, Windows, whatever) using a switch network instance?
I wonder if for those we should explore whether we can detect from the host whether the vnic in the guest has a different MTU setting. Does that get carries across somehow?

Reply:
> Is it correct that the issue appears for a VM (Linux, Windows, whatever) using a switch network instance?
Yes, correct.
> I wonder if for those we should explore whether we can detect from the host whether the vnic in the guest has a different MTU setting. Does that get carries across somehow?
I'm not aware of anything, will investigate...

Reply:
For now we can just document that it is up to the user to make sure that the VMs are configured with jumbo-frames in addition to the network.

Reply:
Updated documentation to explain all these cases and how user should deal with them.